test/suite/test_tiered15.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# test_tiered15.py
+#   Test the "type" configuration in session.create with tiered storage.
+
+from helper_tiered import generate_s3_prefix, get_auth_token, get_bucket1_name, TieredConfigMixin, tiered_storage_sources
+from wtscenario import make_scenarios
+import wiredtiger, wttest
+
+class test_tiered15(TieredConfigMixin, wttest.WiredTigerTestCase):
+    uri_a = "table:tiereda"
+    uri_b = "table:tieredb"
+    uri_c = "table:tieredc"
+    uri_d = "table:tieredd"
+    uri_e = "table:tierede"
+    uri_f = "table:tieredf"
+    uri_g = "table:tieredg"
+    uri_h = "table:tieredh"
+
+    nt_uri_a = "table:nontiereda"
+    nt_uri_b = "table:nontieredb"
+    nt_uri_c = "table:nontieredc"
+    nt_uri_d = "table:nontieredd"
+    nt_uri_e = "table:nontierede"
+    nt_uri_f = "table:nontieredf"
+    nt_uri_g = "table:nontieredg"
+    nt_uri_h = "table:nontieredh"
```

Comment:
The whole point of having `make_scenarios` is to avoid this. You should have a scenario for each `type=` that you want to test. See something like `test_alter02.py` for one example. Your entry could look something like:
`('lsm', dict(type='lsm', tiered_err=True, non_tiered_err=False,)),`
`('tier', dict(type='tier', tiered_err=True, non_tiered_err=True)),` ...

If you need more items in the dict such as the error message, add that too. Then the whole function is only a few lines long.

Reply:
Thanks for the feedback, I'll attempt to restructure the test